exchanges/ftx/ftx_test.go

```diff
@@ -0,0 +1,1007 @@
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.Skip()
+	}
+	_, err := f.GetPublicOptionsTrades(time.Time{}, time.Time{}, "5")
+	if err != nil {
+		t.Error(err)
+	}
+	_, err = f.GetPublicOptionsTrades(time.Unix(1559881511, 0), time.Unix(1559901511, 0), "5")
+	if err != nil {
+		t.Error(err)
+	}
+	_, err = f.GetPublicOptionsTrades(time.Unix(1559901511, 0), time.Unix(1559881511, 0), "5")
+	if err == nil {
+		t.Error(err)
+	}
+}
+
+func TestGetOptionsFills(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.Skip()
+	}
+	_, err := f.GetOptionsFills(time.Time{}, time.Time{}, "5")
+	if err != nil {
+		t.Error(err)
+	}
+	_, err = f.GetOptionsFills(time.Unix(1559881511, 0), time.Unix(1559901511, 0), "5")
+	if err != nil {
+		t.Error(err)
+	}
+	_, err = f.GetOptionsFills(time.Unix(1559901511, 0), time.Unix(1559881511, 0), "5")
+	if err == nil {
+		t.Error(err)
+	}
+}
+
+func TestUpdateOrderbook(t *testing.T) {
+	t.Parallel()
+	cp := currency.NewPairWithDelimiter(currency.BTC.String(), currency.USDT.String(), "/")
+	_, err := f.UpdateOrderbook(cp, asset.Spot)
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestUpdateTicker(t *testing.T) {
+	t.Parallel()
+	cp := currency.NewPairWithDelimiter(currency.BTC.String(), currency.USDT.String(), "/")
+	_, err := f.UpdateTicker(cp, asset.Spot)
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestGetActiveOrders(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.Skip("API keys required but not set, skipping test")
+	}
+	var orderReq order.GetOrdersRequest
+	cp := currency.NewPairWithDelimiter(currency.BTC.String(), currency.USDT.String(), "/")
+	orderReq.Pairs = append(orderReq.Pairs, cp)
+	_, err := f.GetActiveOrders(&orderReq)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestGetOrderHistory(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.Skip("API keys required but not set, skipping test")
+	}
+	var orderReq order.GetOrdersRequest
+	cp := currency.NewPairWithDelimiter(currency.BTC.String(), currency.USDT.String(), "/")
+	orderReq.Pairs = append(orderReq.Pairs, cp)
+	_, err := f.GetOrderHistory(&orderReq)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestUpdateAccountHoldings(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.Skip("API keys required but not set, skipping test")
+	}
+	_, err := f.UpdateAccountInfo()
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestFetchAccountInfo(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.Skip("API keys required but not set, skipping test")
+	}
+	_, err := f.FetchAccountInfo()
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestGetFee(t *testing.T) {
+	var x exchange.FeeBuilder
+	x.PurchasePrice = 10
+	x.Amount = 1
+	x.IsMaker = true
+	var a float64
+	var err error
+	if areTestAPIKeysSet() {
+		a, err = f.GetFee(&x)
+		if err != nil {
+			t.Error(err)
+		}
+		if a != 0.0039 {
+			t.Errorf("incorrect maker fee value")
+		}
+	}
+	x.IsMaker = false
+	if areTestAPIKeysSet() {
```

Comment:
Given that these are dynamic, I think it would be good to pass so long as getfee does not return an error and the fee is not 0.

Reply:
sorry can u explain that differently? not understanding it properly

Reply:
When you have API keys set, you then get the fees from an account. That account can have different fee levels. For example, mine fails this test because my fees are at `0.004` and `0.009`. So when api keys are set, and returned fee is > 0 && <= 0.009, then pass as it means ftx has returned a proper value.

Reply:
bump

Reply:
If this is not being changed it might be worth adding a comment to the user explaining for them to update the fee's to match their current level if they are doing live testing

or even change them to user configurable variables